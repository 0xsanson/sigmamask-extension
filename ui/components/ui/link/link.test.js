/* eslint-disable jest/require-top-level-describe */
import { render } from '@testing-library/react';
import React from 'react';

import Link from './link';

const METAMASK_URL = 'https://metamask.io/';
describe('Link', () => {
  it('should render link element correctly with id', () => {
    const { container } = render(<Link id="test-link">Link</Link>);
    expect(container).toMatchSnapshot();
  });

  it('should render link element correctly with href', () => {
    const { container } = render(
      <Link id="test-link" href={METAMASK_URL}>
        Link
      </Link>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render link element correctly with target', () => {
    const { container } = render(
      <Link id="test-link" href={METAMASK_URL} target="_self">
        Link
      </Link>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render link element correctly with res', () => {
    const { container } = render(
      <Link id="test-link" href={METAMASK_URL} rel="license">
        Link
      </Link>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render link element correctly with referer', () => {
    const { container } = render(
      <Link id="test-link" href={METAMASK_URL} rel="license" referer>
        Link
      </Link>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render link element correctly target rel referer', () => {
    const { container } = render(
      <Link
        id="test-link"
        href={METAMASK_URL}
        rel="nofollow"
        target="self"
        referer
      >
        Link
      </Link>,
    );
    expect(container).toMatchSnapshot();
  });
});
