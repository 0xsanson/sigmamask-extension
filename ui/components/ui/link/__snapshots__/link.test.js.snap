// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render link element correctly fully controlled 1`] = `
<div>
  <a
    href="https://metamask.io/"
    id="test-link"
    rel="noopener nofollow"
    target="self"
  >
    Link
  </a>
</div>
`;

exports[`Link should render link element correctly with href 1`] = `
<div>
  <a
    href="https://metamask.io/"
    id="test-link"
    rel="noopener noreferrer"
    target="_blank"
  >
    Link
  </a>
</div>
`;

exports[`Link should render link element correctly with id 1`] = `
<div>
  <a
    id="test-link"
    rel="noopener"
  >
    Link
  </a>
</div>
`;

exports[`Link should render link element correctly with referer 1`] = `
<div>
  <a
    href="https://metamask.io/"
    id="test-link"
    rel="noopener license"
    target="_blank"
  >
    Link
  </a>
</div>
`;

exports[`Link should render link element correctly with res 1`] = `
<div>
  <a
    href="https://metamask.io/"
    id="test-link"
    rel="noopener license noreferrer"
    target="_blank"
  >
    Link
  </a>
</div>
`;

exports[`Link should render link element correctly with target 1`] = `
<div>
  <a
    href="https://metamask.io/"
    id="test-link"
    rel="noopener noreferrer"
    target="_self"
  >
    Link
  </a>
</div>
`;
